### ISSUE TITLE
Bug 10: Data Leakage from Shuffling Time Series

### ISSUE THREAD

**Issue by dawdameet** [dawdameet/team-finbugs-fintech#10]
## 🐛 Bug Description
Linear Regression and Random Forest models report unrealistically high R² scores (0.999 or 1.000). The models appear 'perfect', which is impossible for stock prediction.

## ✅ Expected Behavior
Time series data should be split chronologically without shuffling. Past data trains the model, future data tests it.

## ❌ Current Behavior
train_test_split is shuffling past and future data together, causing massive data leakage. The model sees future prices during training.

## 📁 Files Affected
`stock_price_predictor/predictor.py`

## 🎯 Difficulty: MEDIUM
**Points: 20**

## 🔍 Reproduction Steps
```
1. Run the predictor and observe unrealistically high R² scores
2. Examine the prepare_ml_data method
3. Look at the train_test_split call
4. Notice there's no shuffle parameter specified
5. Research that train_test_split shuffles by default
6. Add shuffle=False parameter
```

## 💡 Hints
Add shuffle=False to preserve chronological order

## 🔧 Solution (Hidden)
<details>
<summary>Click to reveal solution</summary>

```
Change 'return train_test_split(X, y, test_size=0.2)' to 'return train_test_split(X, y, test_size=0.2, shuffle=False)'
```
</details>

---
**Bug ID**: 10  
**Module**: FinBugs (Fintech AI)  
**Status**: 🔴 UNRESOLVED
